presto-druid/src/main/java/io/prestosql/plugin/druid/DruidJdbcClient.java

```diff
@@ -0,0 +1,365 @@
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.emptyToNull;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.prestosql.plugin.jdbc.JdbcErrorCode.JDBC_ERROR;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.varcharColumnMapping;
+import static io.prestosql.plugin.jdbc.TypeHandlingJdbcSessionProperties.getUnsupportedTypeHandling;
+import static io.prestosql.plugin.jdbc.UnsupportedTypeHandling.IGNORE;
+import static io.prestosql.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.prestosql.spi.type.VarcharType.createVarcharType;
+import static java.lang.String.format;
+import static java.sql.DatabaseMetaData.columnNoNulls;
+
+public class DruidJdbcClient
+        extends BaseJdbcClient
+{
+    // Druid maintains its datasources related metadata by setting the catalog name as "druid"
+    // Note that while a user may name the catalog name as something else, metadata queries made
+    // to druid will always have the TABLE_CATALOG set to DRUID_CATALOG
+    private static final String DRUID_CATALOG = "druid";
+    // All the datasources in Druid are created under schema "druid"
+    public static final String DRUID_SCHEMA = "druid";
+
+    @Inject
+    public DruidJdbcClient(BaseJdbcConfig config, ConnectionFactory connectionFactory)
+            throws SQLException
+    {
+        super(config, "\"", connectionFactory);
+    }
+
+    @Override
+    public List<SchemaTableName> getTableNames(JdbcIdentity identity, Optional<String> schema)
+    {
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            try (ResultSet resultSet = getTables(connection, schema, Optional.empty())) {
+                ImmutableList.Builder<SchemaTableName> list = ImmutableList.builder();
+                while (resultSet.next()) {
+                    list.add(new SchemaTableName(
+                            resultSet.getString("TABLE_SCHEM"),
+                            resultSet.getString("TABLE_NAME")));
+                }
+                return list.build();
+            }
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    protected Collection<String> listSchemas(Connection connection)
+    {
+        return ImmutableList.of(DRUID_SCHEMA);
+    }
+
+    /**
+     * Overriden to filter out tables that don't match @param schemaTableName
+     */
```

Comment:
```suggestion
    // Overridden to filter out tables that don't match schemaTableName
```

(typoe, plus it's not a javadoc)